an-rope/src/lib.rs

```diff
@@ -424,25 +425,26 @@ impl Rope {
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from(String::from("efgh"));
-    /// an_rope.prepend(Rope::from(String::from("abcd")))
-    /// assert_eq!(an_rope, "abcdefgh");
+    /// an_rope.prepend(Rope::from(String::from("abcd")));
+    /// assert_eq!(&an_rope, "abcdefgh");
     /// ```
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from("");
     /// an_rope.prepend(Rope::from("abcd"));
-    /// assert_eq!(an_rope, "abcd");
+    /// assert_eq!(&an_rope, "abcd");
     /// ```
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from("abcd");
     /// an_rope.prepend(Rope::from(""));
-    /// assert_eq!(an_rope, "abcd");
+    /// assert_eq!(&an_rope, "abcd");
```

Comment:
I put these tests there to ensure that the empty rope wasn't an edge case that caused Weird Behaviour. Considering that we also have tests for the non-trivial case, I think they should stay.